core/modules/catalog/helpers/optionLabel.ts

```diff
@@ -20,9 +20,15 @@ export function optionLabel (state, { attributeKey, searchBy = 'code', optionId
 
   let attr = state['list_by_' + searchBy][attributeKey]
   if (attr) {
-    let opt = attr.options.find((op) => toString(op.value) === toString(optionId))
+    let opt = {
+      'label': null
+    }
+
+    if (attr.options) {
```

Comment:
You could remove this whole condition as all is done in lines 22 and 23

Reply:
Hi Fifciuu, some commits were wrong, i have rewrited the history with your suggestions. Can you tell me if now is okay? Thank you.

Reply:
Lines 23-36 should be replaced with:
```js
let opt = attr.options.find(option => toString(option.value) === toString(optionId));

if (opt && opt.label) {
  if (!state.labels[attributeKey]) {
    state.labels[attributeKey] = {}
  }
  state.labels[attributeKey][optionId] = opt.label
  return opt.label
```